src/Swift.Bindings/src/Emitter/StringEmitter/Handler/TypeHandler.cs

```diff
@@ -519,16 +519,120 @@ IntPtr ISwiftObject.MarshalToSwift(IntPtr swiftDest)
         /// </summary>
         private void WriteGetProtocolConformanceDescriptor()
         {
+            var libPath = _typeDatabase.GetLibraryPath(_moduleDecl.Name);
             var text = $$"""
-            static ProtocolConformanceDescriptor ISwiftObject.GetProtocolConformanceDescriptor<U>()
+            static ProtocolConformanceDescriptor ISwiftObject.GetProtocolConformanceDescriptor<TProtocol>()
+                where TProtocol : class
             {
-                throw new NotImplementedException();
+                var dispatch = new Dictionary<Type, string>
+                {
+                    {{GenerateGetProtocolConformanceDictionaryEntries()}}
+                };
+
+                if (!dispatch.ContainsKey(typeof(TProtocol)))
+                {
+                    throw new SwiftRuntimeException($"Attempted to retrieve protocol conformance descriptor for type {{_structDecl.Name}} and protocol {typeof(TProtocol).Name}, but no conformance was found.");
```

Comment:
The second string parameter needs double `{{`.

Reply:
Maybe eventually this should turn into a C# based message - since the mistake is on the C# side. But it's not very important now.

Reply:
This is actually correct, the substitution of the second string parameter happens during compilation of projections

Reply:
Right - sorry - this is too "meta" for me :-)